hkmc2/shared/src/main/scala/hkmc2/semantics/Term.scala

```diff
@@ -346,6 +347,15 @@ object PlainFld:
 final case class Spd(eager: Bool, term: Term) extends Elem:
   def showDbg: Str = (if eager then "..." else "..") + term.showDbg
 
+/** 
+ * Contextual Arguments: to be filled in by the implicit resolver.
+ */
+abstract class CtxArg extends Elem:
+  def term: Opt[Term]
+  def showDbg: Str = s"using ${term.fold("‹unpopulated›")(_.showDbg)}"
+
+object CtxArg:
+  def unapply(arg: CtxArg): Opt[Term] = arg.term
```

Comment:
What is this extractor for? It relies on the object's state, which could be confusing/a source of bugs.